File: 5-longest-palindromic-substring/longest-palindromic-substring.ts
function longestPalindrome(s: string): string {
    let res: string = "";
    let maxLength: number = 0;

    for (let i = 0; i < s.length; i++) {
        let left: number = i;
        let right: number = i;
        while (left >= 0 && right < s.length && (s[left] === s[right])) {
            const currLength: number = right - left + 1;
            if (currLength > maxLength) {
                res = s.slice(left, right + 1);
                maxLength = currLength;
            }

            left--;
            right++;
        }

        left = i;
        right = i + 1;
        while (left >= 0 && right < s.length && (s[left] === s[right])) {
            const currLength: number = right - left + 1;
            if (currLength > maxLength) {
                res = s.slice(left, right + 1);
                maxLength = currLength;
            }

            left--;
            right++;
        }
    }

    return res;
};

File: 5-longest-palindromic-substring/longest-palindromic-substring.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        res = ""
        maxLength = 0

        for i in range(len(s)):
            left, right = i, i
            while left >= 0 and right < len(s) and (s[left] == s[right]):
                currLength = right - left + 1
                if currLength > maxLength:
                    res = s[left: right + 1]
                    maxLength = currLength

                left -= 1
                right += 1
            
            left, right = i, i + 1
            while left >= 0 and right < len(s) and (s[left] == s[right]):
                currLength = right - left + 1
                if currLength > maxLength:
                    res = s[left: right + 1]
                    maxLength = currLength
                
                left -= 1
                right += 1
            
        return res